### ISSUE TITLE
Repeating API Calls in app deployed to Google Play Store

### ISSUE THREAD

**Issue by WellDunDun** [FlutterFlow/flutterflow-issues#883]
### Has your issue been reported?

- [X] I have searched the existing issues and confirm it has not been reported.
- [X] I give permission for members of the FlutterFlow team to access and test my project for the sole purpose of investigating this issue.

### Current Behavior

I've been trying to get help with this from FlutterFlow support but they couldn't figure it out. The app behaves differently when deployed to the Google Play Store compared to the web test modes and Android Studio through an emulator. 

I'm not sure how to explain this but I have some on page load API calls that are repeating so many time it's hammering the backend server. It's even repeating calls from a previous page when you navigate to a new page (maybe backed up calls because it's triggering so many?). All of the calls are repeating and one in particular more than others. I don't know exactly how to describe this any better so I created a short loom to show what's happening in the backend when accessing the app from a mobile device through the google play store. (see attachment)

### Expected Behavior

I expect the API call to only happen once 

### Steps to Reproduce

1. Create an app with API Calls
2. Deploy app to Google Play Store
3. Download and run app

### Reproducible from Blank

- [X] The steps to reproduce above start from a blank project.

### Bug Report Code (Required)

IT4wj8/14YZ1ochE0br+bMF7/WI4GDshU7g31OtEESs8GqLuPe0UWMzBUBFUcMu8dGxqMEyJnXMX/MrmkoTTFsI5EzKuQKJgzqtqYQzvRl6gV5StEK63b3B5BcNQfmaG4JiWnhVSKrNsdCUl6173B+ivYHTDO6a0DDA/GOaeXdeK2SrDX1iXc2URm05KZDPz

### Context

I can't launch my app until this is fixed. My goal is to use Xano as my backend like I do with so many other frontends but have constantly run into issues.

### Visual documentation

https://www.loom.com/share/6488c575d0514cdfa92d0508b066ec87

### Additional Info

The bug report code is for the categories API call shown in the video.

### Environment

```markdown
- FlutterFlow version: v3.1 - Flutter 3.7.12
- Platform: Android
- Browser name and version:
- Operating system and version affected:
```

**Comment by bulgariamitko** [comment]
This is normal FF behavior . If you want the API to only run once you need to implement caching.

**Comment by WellDunDun** [comment]
> This is normal FF behavior . If you want the API to only run once you need to implement caching.

That's very weird for normal behavior. I've implemented caching and it still does it on some calls. 

Do you know why this happens on a deployed app but not on Android Studio or in the web Run/Test modes?

Are you talking about caching through the API Calls menu then under the Advanced Settings tab enable "Cache API Results"? If so, how can I avoid having stale data with that option if we can't set a TTL or use actions to clear the query cache like you can with Backend Query Cache?

Thanks for your help, Dimitar!

**Comment by bulgariamitko** [comment]
> Do you know why this happens on a deployed app but not on Android Studio or in the web Run/Test modes?

When you say it is not happening on Run/Test mode how did you know it is not happening, did you check the Network panel? 

You can use actions to clear the cache. There is an action to clear the cache.

**Comment by WellDunDun** [comment]
> > Do you know why this happens on a deployed app but not on Android Studio or in the web Run/Test modes?
> 
> When you say it is not happening on Run/Test mode how did you know it is not happening, did you check the Network panel?
> 
> You can use actions to clear the cache. There is an action to clear the cache.

Yes, I was checking the Network panel and my Xano server request history.

There is no action to clear the cache for an API Call that is done on page load - only if it is done as a backend query on the widget tree.

**Comment by bulgariamitko** [comment]
@WellDunDun , there is 
![whats-new2-FlutterFlow (1)](https://github.com/FlutterFlow/flutterflow-issues/assets/7221384/b3b3b4d2-39ea-447f-8d0e-07579023a635)

**Comment by bulgariamitko** [comment]
@WellDunDun , Yes I understand what you are saying. To fix that you can still query your API calls in the Widget and then store the data in an App State and thne use this App State to populate your data. Are you using the result of the data in the On Page load actions if not then you can use Widget state query directly.

**Comment by WellDunDun** [comment]
@bulgariamitko Hey Dimitar, do you have any other ideas on how I can solve this? I've noticed other people reporting the same behavior even with cached API calls.

After doing more digging, I found [this article ](https://flutterigniter.com/future-async-called-multiple-times/)and think that the code is invoking another callAsyncFetch() before the cache is set. Thoughts?

**Comment by bulgariamitko** [comment]
@WellDunDun , you can try to write that in a custom code, you can take the API from the FF source code 

Here is a code example 
https://github.com/bulgariamitko/flutterflowtutorials/blob/0e0dde9eec21d9ef1a679074155c2ef6f849bea3/Custom%20Actions/manage-stripe-subs.dart#L6

**Comment by PauloAlves311** [comment]
Are the results coming from an infinite scroll list?

**Comment by aemelyanovff** [comment]
@WellDunDun Sorry to hear about this issue. This issue tracker is intended for general bugs that are applicable to all users. Because this error appears to be specific to your project and the steps are not reproducible, this is classified as a support request and will be closed.

**Comment by WellDunDun** [comment]
@aemelyanovff @bulgariamitko I have found the issue! It is tied to how the FF team built the call using Future. You must take the Future out of this build method so it doesn't refire once for every call to the parent’s build. [This article](https://flutterigniter.com/future-async-called-multiple-times/) goes over it and shows multiple ways to fix this.

**Comment by WellDunDun** [comment]
@aemelyanovff @bulgariamitko Can y'all reopen the ticket or should I create a new one?

**Comment by aemelyanovff** [comment]
@WellDunDun You are right that Futures for `FutureBuilder` _in general_ should be initialized in `initState()`. But it's not always possible, so instead FlutterFlow implements query caching which should prevent duplicate API/DB calls.

If you enabled query caching but you are still observing duplicate API calls, please open a new issue, but please include **specific** steps to reproduce, starting from a blank project.
